h/security/policy/_api_cookie.py

```diff
@@ -0,0 +1,34 @@
+from pyramid.request import Request
+from pyramid.security import Allowed, Denied
+
+from h.security.identity import Identity
+from h.security.policy._cookie import CookiePolicy
+
+COOKIE_AUTHENTICATABLE_API_REQUESTS = [
+    ("api.groups", "POST"),  # Create a new group.
+    ("api.group", "PATCH"),  # Edit an existing group.
+]
+
+
+class APICookiePolicy:
+    """Authenticate API requests with cookies."""
+
+    def __init__(self, cookie_policy: CookiePolicy):
+        self.cookie_policy = cookie_policy
+
+    @staticmethod
+    def handles(request: Request) -> bool:
+        """Return True if this policy applies to `request`."""
+        return (
+            request.matched_route.name,
+            request.method,
+        ) in COOKIE_AUTHENTICATABLE_API_REQUESTS
+
+    def identity(self, request: Request) -> Identity | None:
+        return self.cookie_policy.identity(request)
+
+    def authenticated_userid(self, request: Request) -> str | None:
+        return self.cookie_policy.authenticated_userid(request)
+
+    def permits(self, request: Request, context, permission: str) -> Allowed | Denied:
+        return self.cookie_policy.permits(request, context, permission)
```

Comment:
I went with a separate `APICookiePolicy` class rather than having `APIPolicy` just delegate to `CookiePolicy` itself for a couple of reasons:

1. The `handles()` method is used by `APIPolicy` to decide when to delegate to `APICookiePolicy`, but `TopLevelPolicy` does not use `handles()` when delegating non-API requests to `CookiePolicy`, it just [blindly delegates all non-API requests to `CookiePolicy`](https://github.com/hypothesis/h/blob/fad818020e426f236bb991d3a6e1332b653892f6/h/security/policy/top_level.py#L33-L39). I thought it'd be weird for `CookiePolicy` to have a `handles()` method that's only used in the context of API requests
2. API sub-policies don't need the `remember()` and `forget()` methods and `APIPolicy` never calls these methods, so it's potentially a bit confusing for an API sub-policy to have these methods.

Using composition rather than inheritance here which I think leads to much clearer code and better, more isolated unit tests. Also allows `APICookiePolicy` to use `CookiePolicy` for `identity()`, `authenticated_userid()` and `permits()` but not inherit `CookiePolicy`'s `remember()` and `forget()`. Also `APICookiePolicy.__init__()` will not be affected by any changes to `CookiePolicy.__init__()`